var/spack/repos/builtin.mock/packages/hg-top-level/package.py

```diff
@@ -14,4 +14,5 @@ class HgTopLevel(Package):
     version('1.0')
 
     def install(self, spec, prefix):
-        pass
+        # sanity_check_prefix requires something in the install directory
+        mkdirp(prefix.bin)
```

Comment:
rather than writing this in every test package, can't you either:

1. Make a `TestPackage` that has this install method in it and just have all the test packages extend it
2. `monkeypatch` this install method into `Package` in the `install_mockery` fixture?

then at least the dummy install method would be written in (mostly) one place.  Some packages (e.g., `HashTest3`) would obviously override it.

Reply:
I remember you suggesting #1 before (and now regret not taking you up on that at that time) :).  I could for all of the test packages that extend `Package` or `PackageBase`.  IIRC not all affected do but I'll check as perhaps they should.

Reply:
Tried #1 two ways so the need for `install` was explicit.  Putting it in `package.py` worked but is not "clean" from the perspective of separating test features from the core code.  

The second approach involved creating a `TestPackage` in the `builtin.mock` repo and having packages inherit from it.  This approach causes mirror test failures and test hangs (after renaming the package to `Testpackage`).  Turns out this approach is not (yet) supported in Spack.

Commit `2293fe0` uses `monkeypatch` on `PackageBase` with the handy option of not raising an exception if the `install` attribute is not defined.